SCons/Node/FS.py

```diff
@@ -255,7 +255,7 @@ def _softlink_func(fs, src, dst):
 def _copy_func(fs, src, dest):
     shutil.copy2(src, dest)
     st = fs.stat(src)
-    fs.chmod(dest, stat.S_IMODE(st[stat.ST_MODE]) | stat.S_IWRITE)
+    fs.chmod(dest, stat.S_IMODE(st.st_mode) | stat.S_IWRITE)
```

Comment:
Was this previously incorrect? 
Can you explain this change? Not sure I understand it.

Reply:
stat now returns a stat object, we can just use the attribute, although the old way of indexing still works.  I did this replacement where it's like-for-like; I didn't replace the one for ST_MTIME because that changed the retrieved value from int to float which caused problems (fixable, but I backed it out in a previous commit to keep it simpler)

Reply:
Gotcha. Thanks!